deployments/puppet/manifests/init.pp

```diff
@@ -405,10 +408,35 @@
         ensure => absent,
       }
       if $auto_instrumentation_version == 'latest' or versioncmp($auto_instrumentation_version, '0.87.0') >= 0 {
-        file { $zeroconfig_java_config_path:
-          ensure  => file,
-          content => template('splunk_otel_collector/java.conf.erb'),
-          require => Package[$auto_instrumentation_package_name],
+        if 'java' in $with_auto_instrumentation_sdks {
+          file { $zeroconfig_java_config_path:
+            ensure  => file,
+            content => template('splunk_otel_collector/java.conf.erb'),
+            require => Package[$auto_instrumentation_package_name],
+          }
+        }
+        if 'nodejs' in $with_auto_instrumentation_sdks {
+          $splunk_otel_js_path = '/usr/lib/splunk-instrumentation/splunk-otel-js.tgz'
+          $splunk_otel_js_prefix = '/usr/lib/splunk-instrumentation/splunk-otel-js'
+
+          exec { 'Check npm is present':
+            command  => "bash -c 'command -v ${$auto_instrumentation_npm_path}'",
+            provider => shell,
+          }
+
+          file { "${$splunk_otel_js_prefix}/node_modules":
+            ensure => 'directory',
+          }
+          exec { 'Install splunk-otel-js':
+            command  => "${$auto_instrumentation_npm_path} install ${$splunk_otel_js_path}",
```

Comment:
Looking at puppet's [resource relationship](https://www.puppet.com/docs/puppet/5.5/lang_relationships.html) doc, this resource will be executed even if npm is not installed. We should only execute if npm exists. Also wondering if puppet has an `npm` provider we should use to install packages.

Reply:
I assumed line 422 would trigger first. I think we can add an explicit test for this as a way to prove this works as it should.

Reply:
It would, and puppet would mark that resource as failed but that does not block execution of other resources. So we'll have this one also marked as failed. That's what I gather from their docs. I think what we want is to mark a the resource which checks npm is not installed as failed and then skip any resources which are trying to setup the nodejs stuff. The user can then notice the failed npm resource and decide on either installing npm to make it work or disable nodejs explicitly.

Reply:
I have removed the check because we want to make sure the application of puppet actually fails, and all the safeguards would silently pass on the absence of npm, leaving users without notice that installation didn't happen.